tap_facebook/__init__.py

```diff
@@ -278,22 +280,38 @@ def run_job(self, params):
                     'Insights job {} did not start after {} seconds'.format(
                         job_id, INSIGHTS_MAX_WAIT_TO_START_SECONDS))
             elif duration > INSIGHTS_MAX_WAIT_TO_FINISH_SECONDS and status != "Job Completed":
-                pretty_error_message = ('Insights job {} did not complete after {} minutes. ' +
-                    'This is an intermittent error and may resolve itself on subsequent queries to the Facebook API. ' +
-                    'You should deselect fields from the schema that are not necessary, ' +
-                    'as that may help improve the reliability of the Facebook API.')
-                raise Exception(pretty_error_message.format(job_id, INSIGHTS_MAX_WAIT_TO_FINISH_SECONDS//60))
+                base_message = 'Insights job {} did not complete after {} minutes. '.format(job_id,
+                                                                                            INSIGHTS_MAX_WAIT_TO_FINISH_SECONDS//60)
+                if self.can_retry:
+                    LOGGER.info(base_message + 'Retrying...')
+                    raise InsightsJobTimeout()
+                else:
+                    pretty_error_message = (base_message +
+                                            'This is an intermittent error and may resolve itself on subsequent queries to ' +
+                                            'the Facebook API. You should deselect fields from the schema that are not ' +
+                                            'necessary, as that may help improve the reliability of the Facebook API.')
+                    raise Exception(pretty_error_message.format(job_id, INSIGHTS_MAX_WAIT_TO_FINISH_SECONDS//60))
```

Comment:
Is the call to `format` here necessary? It seems like we're already doing that up on line 283.

Reply:
Nope! Just forgot to remove it, good catch.